### ISSUE TITLE
Support attaching files to open test reports

### ISSUE THREAD

**Issue by mpkorstanje** [cucumber/cucumber-jvm#2947]
### 🤔 What's the problem you're trying to solve?

The JUnit Team is working on a new reporting format that more accurately represents hierarchical tests (https://github.com/junit-team/junit5/issues/4113).  Recently support for file attachments was added. However, while Cucumber already supports attaching files to reports, it does not yet translate these to the JUnit Platform.

### ✨ What's your proposed solution?

* Map the embed and log events to JUnits file and report entries respectively.
* Request support for media/mime types in reports from open test reporting
* Request support for log files in reports from open test reporting

**Comment by mpkorstanje** [comment]
I as framework maintainer have no need for it. But as a consumer of test reports I like them to be richly formatted. The media type allows for a rich representation of the attached file without an externalized file. I imagine that the OTA based reports could also have such a rich representation.

One example be the [cucumber-html report](https://github.com/cucumber/react-components?tab=readme-ov-file#attachments) report. 

But a better, more impressive, example would be Allure. It supports [several media types](https://allurereport.org/docs/attachments/#show-overlay). Including a [`application/vnd.allure.image.diff`](https://allurereport.org/docs/attachments/#show-overlay) which would be hard to replicate by guessing the media type based on a file extension.

Allures API for attaching content to a report looks like this:

```java
...
        // Attach to the test report
        Allure.addAttachment("Screenshot diff",
                "application/vnd.allure.image.diff",
                content);
...
```                

And I thinking out loud, it would be useful if JUnit had a something similar that would attach the bytes to the report, potentially externalizing them into a file.

**Comment by mpkorstanje** [comment]
And thinking out loud some more, another useful feature for OTA would be the ability to report test steps. For Cucumber this would allow the steps in a scenario to be represented in the OTA report. And these are also present in Allure. https://allurereport.org/docs/attachments/#show-overlay

And with that it comes to mind that to drive adoption of Open Test Reporting format, you may also want to talk to tools that create test reports.

**Comment by marcphilipp** [comment]
> And thinking out loud some more, another useful feature for OTA would be the ability to report test steps. For Cucumber this would allow the steps in a scenario to be represented in the OTA report.

This could be done in a (Cucumber-specific) extension schema.

**Comment by mpkorstanje** [comment]
> This could be done in a (Cucumber-specific) extension schema.

Not when used as a JUnit Platform Engine though. Unless I've missed something.

**Comment by marcphilipp** [comment]
That's right. We'd have to extend `TestExecutionListener` to make custom events possible and somehow make the listener implementation that writes the XML file extensible. :thinking:
